ui/src/App.jsx

```diff
@@ -22,20 +22,28 @@ class App extends Component {
 
   refreshState() {
     axios
-      .get(`${window.config.env.REACT_APP_NOPAS_ENDPOINT}/state`)
+      .get(`${window.config.env.REACT_APP_NOPAS_ENDPOINT}/predefined`)
       .then(response => {
-        try {
-          const newState = response.data;
-          for (let sp of newState.Subpolicies) {
-            // convert metadata object to string
-            sp.Metadata = JSON.stringify(sp.Metadata);
-          }
-          this.props.refreshState(newState);
-        } catch (error) {
-          alert(error);
-        }
+        this.props.updatePossibleDefaultsList(response.data);
+        axios
+          .get(`${window.config.env.REACT_APP_NOPAS_ENDPOINT}/state`)
+          .then(response => {
+            try {
+              const newState = response.data;
+              for (let sp of newState.Subpolicies) {
+                // convert metadata object to string
+                sp.Metadata = JSON.stringify(sp.Metadata);
+              }
+              this.props.refreshState(newState);
+            } catch (error) {
+              alert(error);
+            }
+          })
+          .catch(function(error) {
+            alert(error);
+          });
       })
-      .catch(function(error) {
+      .catch(error => {
         alert(error);
       });
   }
```

Comment:
This is a good place to use `async refreshState()`, then all your `.then` can be replaced with `const xxx = await axios.xxx`. Consequently, you should only need one overarching `try-catch` block.